### PR TITLE
Add "allow uts/ipc/user ns" option (from sylabs #2981)

### PR DESCRIPTION
**Pull request by JasonYangShadow**
## Description of the Pull Request (PR):

This PR re-implements the Singularity PR: https://github.com/sylabs/singularity/pull/2981
This PR re-implements the Singularity PR: https://github.com/sylabs/singularity/pull/2915
This PR re-implements the Singularity PR: https://github.com/sylabs/singularity/pull/2956

which had an original description of
> singularity.conf now accepts setting the allow uts ns option, and can
invalidate the use of the --uts and --hostname flags.

> Add new option to singularity.conf: "allow ipc ns"

> Add new option to singularity.conf: "allow user ns"
Added e2e tests, making sure to fail when allow user ns = no in cases of any of the following flags:
--userns
--fakeroot
Also fails when allow user ns = no in singularity.conf and singularity was installed with the --without-suid option. This last one was checked manually.

### This fixes or addresses the following GitHub issues:

Addresses multiple of the PRs in

- https://github.com/apptainer/apptainer/issues/2546

#### Before submitting a PR, make sure you have done the following:

- [ ] Read the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] Make sure all commits are signed-off with `git commit -s`, see the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md).
- [ ] Added changes to the [CHANGELOG](https://github.com/apptainer/apptainer/blob/main/CHANGELOG.md), if necessary according to the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md).
- [ ] Added tests to validate this PR, linted with `make check`  and tested this PR locally with a `make test`, and `make testall` if possible (see the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md)).
- [ ] Based this PR against the appropriate branch according to the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md).
- [ ] Added myself as a contributor to the [Contributors File](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTORS.md).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +46,14 @@ For older changes see the [archived Singularity change log](https://github.com/a
 - Add configuration files binding for `--nv` option, these files are added in
   the `nvliblist.conf` that are recognized in the NVIDIA Container Toolkit,
   including files for EGL ICD.
+- `apptainer.conf` now accepts setting the `allow uts ns` option, and can
+  invalidate the use of the `--uts` and `--hostname` flags.
```
Comment (DrDaveD) [inline CHANGELOG.md:50]: This is a duplicate of the message below

File: CHANGELOG.md [hunk 2]
```diff
@@ -46,6 +46,14 @@ For older changes see the [archived Singularity change log](https://github.com/a
...
+- `apptainer.conf` now accepts setting new options regarding namespaces:
+  - `allow ipc ns` : disable the use of the `--ipc` flag.
+  - `allow uts ns` : invalidate the use of the `--uts` and `--hostname` flags.
+- `allow user ns` : disable creation of user namespaces. This will prevent
+  execution of containers with the `--userns` or `--fakeroot` flags, and
+  unprivileged installations of Apptainer.
```
Comment (DrDaveD) [inline CHANGELOG.md:56]: This should be indented a level

File: CHANGELOG.md [hunk 3]
```diff
@@ -46,6 +46,14 @@ For older changes see the [archived Singularity change log](https://github.com/a
 - Add configuration files binding for `--nv` option, these files are added in
   the `nvliblist.conf` that are recognized in the NVIDIA Container Toolkit,
   including files for EGL ICD.
+- `apptainer.conf` now accepts setting the `allow uts ns` option, and can
+  invalidate the use of the `--uts` and `--hostname` flags.
+- `apptainer.conf` now accepts setting new options regarding namespaces:
```
Comment (DrDaveD) [inline CHANGELOG.md:51]: It should be made clear that each of these default to `yes` but when set to `no` they will do the following.

**Review by DrDaveD — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (JasonYangShadow) [reply CHANGELOG.md]: thanks, I've revisited the changelog entries.

**Review by DrDaveD — APPROVED** [review]

**Pull request merged**